Add support for labelIcon, remove duplicate main landmark

seanforyou23 opened the pull request:
Just some small tweaks to the first step in the form, I noticed the helper text goes away once you interact with it, and if you're like me and get it wrong the first time - you'll have to go back and touch/change the form field in some way to see the helper text again.. I moved it to a help popover for the label in case its helpful.. let me know what you think or if there's a better way to display it.

Also, I noticed we have duplicate main regions - given this plugin embeds within another page that surely already contains a main region, I think we can just omit it from here.

<img width="1193" alt="Screen Shot 2022-04-11 at 2 23 15 PM" src="https://user-images.githubusercontent.com/5942899/162804642-ed702fa1-41e8-4f12-8425-a11d15a89f29.png">

seanforyou23 committed:
Add support for labelIcon, remove duplicate main landmark

seanforyou23 committed:
[checkstyle]

seanforyou23 committed:
update labels

mturley commented:
@seanforyou23 do you mind adding a couple screenshots here for @vconzola to review? Some day we'll get a preview build mode in here :)

seanforyou23 committed:
stub out popover and helper text for stateless/stateful imports

prevent form from jumping around vertically
migrated => imported
use better help icon

seanforyou23 commented:
<img width="800" alt="Screen Shot 2022-04-13 at 2 33 28 PM" src="https://user-images.githubusercontent.com/5942899/163247823-eb616b89-ede9-46fa-9e3b-459394a63ed1.png">

seanforyou23 committed:
[cleanup]

seanforyou23 committed:
[fixup-feedback]

mturley committed:
Merge branch 'main' into import-wiz-tweaks

djzager commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
-dev/tmp/
\ No newline at end of file
+dev/tmp/
+.DS_Store
```
ahh. MacOS :sunglasses:

djzager commented on src/components/ImportPage.tsx:
```diff
@@ -23,14 +23,14 @@ const ImportPage: React.FunctionComponent<PipelineWizardPageProps> = ({
     <QueryClientProvider client={queryClient}>
       <NamespaceContext.Provider value={namespace}>
-        <Page>
+        <>
```
For my own edification, what is happening here with this change? a no-op?

seanforyou23 replied:
Ah, because we're embedding this bit of UI within another UI that's already accounting for the page's primary `main` region - the markup generated by the Page component here was creating navigation issues for screen reader users by redundantly adding another main region to the page (also other duplicate/unnecessary markup structures)

djzager commented on src/components/ImportWizard/SourceClusterProjectStep.tsx:
```diff
@@ -64,43 +66,88 @@ export const SourceClusterProjectStep: React.FunctionComponent = () => {
-    validateSourceNamespaceQuery.data?.data.kind === 'Namespace',
-    <div className={formStyles.formHelperText}>Name of the project to be migrated</div>,
-  );
+  const sourceNamespaceFieldProps = getAsyncValidationFieldProps({
+    validating: validateSourceNamespaceQuery.isLoading,
+    valid: validateSourceNamespaceQuery.data?.data.kind === 'Namespace',
+    helperText: <div className={formStyles.formHelperText}>Name of the project to be imported</div>,
+  });
```
If you wanted to add a Popover for this option too, similar to the `oc whoami -t` , `oc project` will tell you the project currently in use. From the `oc` help text:

```
Examples:
  # Switch to 'myapp' project
  oc project myapp

  # Display the project currently in use
  oc project
```

Looking at the screenshot shared it wasn't clear how easy it would be for the user to get this help.

seanforyou23 replied:
<img width="917" alt="Screen Shot 2022-04-18 at 4 38 03 PM" src="https://user-images.githubusercontent.com/5942899/163874329-f891c73a-1efc-4b91-9472-9bac45e4582f.png">

Good call! Styling seems to still be getting tweaked in the host app, so the `<code>` snippet has interesting styling - can fix it up here if need be but I think this does the main thing you're after. Lemme know if you see anything else!

djzager reviewed (commented):
A few non-expert comments.

seanforyou23 committed:
fixup reader suppoort

seanforyou23 committed:
improve helper text

seanforyou23 committed:
Merge branch 'import-wiz-tweaks' of github.com:seanforyou23/crane-ui-plugin into import-wiz-tweaks

djzager commented:
/test all

djzager reviewed (approved):
/lgtm

djzager commented:
Trying to get the openshift bot to show up.

/test all

jmontleon commented:
/test all

jmontleon commented:
/approve

jmontleon commented:
/lgtm

jmontleon commented:
/retest

pranavgaikwad reviewed (approved):
/lgtm

mturley commented:
@vconzola did we ever get content folks from UXD to look at this one?

vconzola commented:
@mturley No, I've been waiting since multiple pipelines aren't in scope for tech preview anymore.

mturley commented:
@vconzola oh... I missed that, does that mean we aren't supporting stage for tech preview? I was also about to add the "test" pipeline (cutover without quiesce), are we not going to support that either?

Was this decided when I was on PTO or am I forgetting a meeting?
